### PR TITLE
DEP: Removes unnecessary identifier_cache from asset_finder

### PR DESCRIPTION
**Pull request by jfkirk**
The identifier cache's usage was nearly identical to using lookup_generic, so this commit removes identifier-keyed caching and modifies anything that uses it.

### CODE REVIEW

File: tests/test_assets.py [hunk 1]
```diff
@@ -591,14 +591,14 @@ def test_consume_asset_as_identifier(self):
     def test_sid_assignment(self):
 
         # This metadata does not contain SIDs
-        metadata = {'PLAY': {'symbol': 'PLAY'},
-                    'MSFT': {'symbol': 'MSFT'}}
+        metadata = {'PLAY': {'symbol': 'PLAY', 'sid': 1},
```
Comment (ssanderson) [inline tests/test_assets.py]: @jfkirk isn't the point of this test to exercise how we handle inputs that **don't** provide sid data?

Comment (jfkirk) [inline tests/test_assets.py]: Yes, and I feel incredibly dense now. Thanks!

**Comment by jfkirk** [comment]
@ssanderson Fixed my silly test mistake.

File: tests/test_assets.py [hunk 2]
```diff
@@ -598,8 +598,9 @@ def test_sid_assignment(self):
         finder = AssetFinder(metadata=metadata, allow_sid_assignment=True)
 
         # Verify that Assets were built
-        play = finder.retrieve_asset_by_identifier('PLAY')
+        play = finder.lookup_generic('PLAY', datetime.now())[0]
```
Comment (ssanderson) [inline tests/test_assets.py]: @jfkirk is there a reason we're calling `lookup_generic` here as opposed to `lookup_symbol`?

Comment (jfkirk) [inline tests/test_assets.py]: Nope, hasty habit. Changing now.

Comment (jfkirk) [inline tests/test_assets.py]: @ssanderson Changed to use lookup_symbol

File: tests/test_assets.py [hunk 3]
```diff
@@ -598,8 +598,9 @@ def test_sid_assignment(self):
         finder = AssetFinder(metadata=metadata, allow_sid_assignment=True)
 
         # Verify that Assets were built
-        play = finder.retrieve_asset_by_identifier('PLAY')
+        play = finder.lookup_symbol('PLAY', datetime.now())
         self.assertEqual('PLAY', play.symbol)
+        self.assertIsNotNone(play.sid)
```
Comment (ssanderson) [inline tests/test_assets.py]: Should we also assert that we assign different sids to `'MSFT'` and `'PLAY'`?

Comment (jfkirk) [inline tests/test_assets.py]: Good idea - done.

**Comment by jfkirk** [comment]
This PR is pending some compatibility tests.

File: zipline/assets/assets.py [hunk 4]
```diff
@@ -544,6 +532,30 @@ def lookup_generic(self,
             self._lookup_generic_scalar(obj, as_of_date, matches, missing)
         return matches, missing
 
+    def map_identifier_list_to_sids(self, index, as_of_date):
```
Comment (jfkirk) [inline zipline/assets/assets.py]: @ssanderson Does this method make sense? Its function is similar to lookup_generic, but also will build the assets that are needed.

**Pull request merged**